### PR TITLE
doc: Add missing docs everywhere

### PR DESCRIPTION
**Pull request by aborgna-q**
Copied what I could from TKET's doc.

There is no math rendering for the op definitions, we'll need to add something like [katex_doc](https://docs.rs/katex-doc/0.1.0/katex_doc/) later.

Also deletes the outdated `tket2ops` feature (#11).

### CODE REVIEW

File: src/circuit_json.rs [hunk 1]
```diff
@@ -47,41 +62,57 @@ pub struct BoxID(uuid::Uuid);
...
         id: BoxID,
-        // 4x4 matrix of complex numbers
+        /// 4x4 matrix of complex numbers
         matrix: [[(f32, f32); 4]; 4],
     },
+    /// Three-qubit operation defined as a unitary matrix.
     Unitary3qBox {
         id: BoxID,
-        // 8x8 matric of complex numbers
+        /// 8x8 matric of complex numbers
```
Comment (ss2165) [inline src/circuit_json.rs:89]:
```suggestion
        /// 8x8 matrix of complex numbers
```

File: src/circuit_json.rs [hunk 2]
```diff
@@ -47,41 +62,57 @@ pub struct BoxID(uuid::Uuid);
...
         phase: String,
     },
+    /// An operation capable of representing arbitrary Circuits made up of CNOT
+    /// and RZ, as a PhasePolynomial plus a boolean matrix representing an
+    /// additional linear transformation.
     PhasePolyBox {
         id: BoxID,
+        /// Number of qubits.
         n_qubits: u32,
+        ///
```
Comment (ss2165) [inline src/circuit_json.rs:115]: ?

File: src/optype.rs [hunk 3]
```diff
@@ -1,594 +1,432 @@
...
     Rotation,
+    ///
     #[cfg(feature = "tket2ops")]
     ToRotation,
+    ///
     #[cfg(feature = "tket2ops")]
     Copy,
+    ///
     #[cfg(feature = "tket2ops")]
     Const,
```
Comment (ss2165) [inline src/optype.rs:431]:
```suggestion

```

**Review by ss2165 — CHANGES_REQUESTED** [review]

File: src/optype.rs (hunk 3 above)
Comment (aborgna-q) [reply src/optype.rs:431]: I put this in a different PR, but may as well do it here.

File: src/circuit_json.rs (hunk 2 above)
Comment (aborgna-q) [reply src/circuit_json.rs:115]:
```suggestion
```
I'll remove the line.

**Review by ss2165 — APPROVED** [review]
:shipit:

**Pull request merged**